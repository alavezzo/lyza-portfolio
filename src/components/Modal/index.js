import React from 'react';

function Modal ({ currentPhoto, onClose }) {
    const {name, description, category, index} = currentPhoto;
    return ( 
        <div className="modalBackdrop">
            <div className="modalContainer">
                <h3 className="modalTitle">{name}</h3>
                <img alt={category} src={require(`../../assets/large/${category}/${index}.jpg`).default}/>
                <p>
                    {description}
                </p>
                <button onClick={onClose}  type ="button">
                    Close this modal
                </button>
            </div>
        </div>
    );

};

export default Modal;