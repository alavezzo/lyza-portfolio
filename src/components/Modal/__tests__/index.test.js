import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

afterEach(cleanup);

const currentPhoto = {
    name: '', 
    category: 'portraits',
    description: 'loremispum',
    index: '0'
}
const mockToggleModal = jest.fn()

describe('Modal Tests', () => {

    it('Modal is rendering', () => {

        render(<Modal 
           currentPhoto={currentPhoto}
           onClose={mockToggleModal}
            />)
    });

    it('Matches Snapshot', () => {
        const { asFragment } = render(<Modal 
            currentPhoto={currentPhoto}
            onClose={mockToggleModal}
             />);
        expect(asFragment()).toMatchSnapshot();
    });

    describe('Click Event', () => {
        it('calls onClose handler', () => {
            // Arrange: Render Modal
            const { getByText } = render(<Modal
                onClose={mockToggleModal}
                currentPhoto = {currentPhoto} />)
            // Act: Simulate click event 
            fireEvent.click(getByText('Close this modal'))
            // Assert: Expected matcher
            expect(mockToggleModal).toHaveBeenCalledTimes(1);
        })
    })

});