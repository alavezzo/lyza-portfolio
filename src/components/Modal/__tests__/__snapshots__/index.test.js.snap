// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Tests Matches Snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      />
      <img
        alt="portraits"
      />
      <p>
        loremispum
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
